blog/2021-09-21-intro-to-isolated-vm.md

```diff
@@ -0,0 +1,163 @@
+---
+tags:
+  - temporal
+  - community
+posted_on_: 2021-09-14T00:00:00Z
+slug: intro-to-isolated-vm
+title: Introduction to isolated-vm in Node.js
+author: Valeri Karpov
+author_title: Community Member
+author_image_url: https://avatars.githubusercontent.com/u/1620265?v=4
+release_version: V1.9.1
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<!--truncate-->
+
+## Introduction
+
+[Temporal's Node SDK](/docs/node/introduction) makes heavy use of the [isolated-vm npm package](https://www.npmjs.com/package/isolated-vm) to ensure your [Workflows are deterministic](/docs/node/determinism).
+Each Workflow runs in a separate _isolate_ that prevents your Workflow code from directly accessing any logic that may break Workflow determinism, like reading from the file system or generating a random number.
+In this blog post, I'll cover why the Node SDK uses isolated-vm, sketch out how the Node SDK uses isolated-vm, and describe what that means for how you write Workflows in Node.js.
+
+## Why Isolates?
+
+The key idea behind Temporal is that each Workflow is stored in the Temporal server as an [_event history_](/blog/temporal-tips-tricks-1/#event-history).
+Instead of storing the "current state" of each Workflow, Temporal instead stores the Workflow's initial state, all Signals the Workflow has received, and the result of all Activities the Workflow executed.
+
+If the Temporal server wants to pause the Workflow and resume it on a different Worker, the Temporal server will replay the Workflow function from the beginning using the same Signals and Activity results.
```

Comment:
Maybe give examples as to why it might do that, e.g. worker evicts workflow from cache or is stopped.

Reply:
Elsewhere in docs, Temporal Server is title case